cpp/marc_tools/single_marc_convert_keywords.cc

```diff
@@ -59,69 +59,65 @@ void ExtractSubfieldsForTag(const MARC::Record &record, const std::string &subfi
 
-    while (MARC::Record record = marc_reader->read()) {
+    while (const MARC::Record record = marc_reader->read()) {
         ++record_count;
 
         std::vector<std::string> subfields;
-        ExtractSubfieldsForTag(record, "150", "agx", &subfields);
-        // making 2 entries: one for every single 'a' subfield and one for all subfields
-        if (subfields.size() == 1)
-            gnd_keywords->emplace(std::make_pair(subfields[0], record.getControlNumber()));
-        ExtractSubfieldsForTag(record, "100", "atg", &subfields);
-        ExtractSubfieldsForTag(record, "110", "agx", &subfields);
-        ExtractSubfieldsForTag(record, "111", "agx", &subfields);
-        ExtractSubfieldsForTag(record, "130", "agx", &subfields);
-        ExtractSubfieldsForTag(record, "151", "agx", &subfields);
-        // second entry for all subfields combined
-        if (subfields.size() > 1) {
-            const std::string key(StringUtil::Join(subfields, " "));
-            gnd_keywords->emplace(std::make_pair(key , record.getControlNumber()));
-        }
+        AddMainSubfieldAndCombinationsToGndKeywords(record, keywords_to_gnd_numbers_map, "150", "agx", &subfields);
+        AddMainSubfieldAndCombinationsToGndKeywords(record, keywords_to_gnd_numbers_map, "100", "atg", &subfields);
+        AddMainSubfieldAndCombinationsToGndKeywords(record, keywords_to_gnd_numbers_map, "110", "agx", &subfields);
+        AddMainSubfieldAndCombinationsToGndKeywords(record, keywords_to_gnd_numbers_map, "111", "agx", &subfields);
+        AddMainSubfieldAndCombinationsToGndKeywords(record, keywords_to_gnd_numbers_map, "130", "agx", &subfields);
+        AddMainSubfieldAndCombinationsToGndKeywords(record, keywords_to_gnd_numbers_map, "151", "agx", &subfields);
     }
     LOG_INFO("Processed " + std::to_string(record_count) + " MARC record(s).");
 }
 
 
 void FindEquivalentKeywords(std::unordered_map<std::string, std::string> const &keywords_to_gnd_numbers_map, std::unordered_set<std::string> const &keywords_to_compare, const std::string matches_output_file, const std::string no_matches_output_file) {
-    std::unordered_map<std::string, std::string> keyword_matches;
+    std::unordered_map<std::string, std::string> keyword_to_ppn_map;
     std::unordered_set<std::string> keywords_without_match;
     for (const auto &keyword : keywords_to_compare) {
         const auto gnd_number(keywords_to_gnd_numbers_map.find(keyword));
         if (gnd_number == keywords_to_gnd_numbers_map.end())
             keywords_without_match.insert(keyword);
         else
-            keyword_matches.insert(*gnd_number);
+            keyword_to_ppn_map.insert(*gnd_number);
     }
-    LOG_INFO("Found " + std::to_string(keyword_matches.size()) + " keyword matches.\n");
-    const double percentage((static_cast<double>(keyword_matches.size())/static_cast<double>(keywords_to_compare.size())) * 100);
+    LOG_INFO("Found " + std::to_string(keyword_to_ppn_map.size()) + " keyword matches.\n");
```

Comment:
Please put parentheses around "es" here as I assume it can happen that the map size is 1.